Refactor/user

mohamedebrahem13 opened the pull request:
- refactor user repo 
- create mappers
- create request classes  
- add serialization
- remove dummy data 
- use fizzed

mohamedebrahem13 committed:
refactor(data): Refactor UserRemoteDataSource and introduce request models

mohamedebrahem13 committed:
feat(user): Migrate User model to Freezed and remove ID field

mohamedebrahem13 committed:
refactor(data): Decouple `UserRepositoryImpl` from storage and introduce request models

mohamedebrahem13 committed:
feat(deps): Update dependencies and add Freezed for code generation

mohamedebrahem13 committed:
refactor(data): Convert StoreModel to Freezed class

mohamedebrahem13 committed:
refactor(user): Update user API endpoints to be context-based

mohamedebrahem13 committed:
refactor(auth): Reorganize auth remote data source

mohamedebrahem13 committed:
feat(data): Add order mapper import

mohamedebrahem13 committed:
refactor(data): Reorganize remote datasource directory structure

mohamedebrahem13 committed:
refactor(data): Migrate AddressModel to Freezed

mohamedebrahem13 committed:
refactor(auth): Remove redundant user ID storage on login/signup

mohamedebrahem13 committed:
refactor(di): Comment out unused repository registrations

mohamedebrahem13 committed:
refactor(models): Migrate OrderModel and ReviewModel to Freezed

mohamedebrahem13 committed:
refactor(data): Migrate ContactInfoModel to Freezed

mohamedebrahem13 committed:
refactor(data): Add missing import for category mapper

mohamedebrahem13 committed:
refactor(data): Migrate CategoryModel to Freezed

mohamedebrahem13 committed:
feat(address): Add `AddAddressRequest` data model

mohamedebrahem13 committed:
feat(address): Add AddressModel to Address entity mapper

mohamedebrahem13 committed:
feat(auth): Add AuthRemoteDataSource abstract class

mohamedebrahem13 committed:
feat(user): Add mapper for UserModel to User entity

mohamedebrahem13 committed:
feat(user): Add user remote data source implementation

mohamedebrahem13 committed:
feat(profile): Add `UpdateUserProfileRequest` data model

mohamedebrahem13 committed:
feat(address): Add update address request model

mohamedebrahem13 committed:
feat(category): Add mapper for Category model

mohamedebrahem13 committed:
feat(auth): Add ChangePasswordRequest model

mohamedebrahem13 committed:
feat(data): Add mappers for Order and ContactType

mohamedebrahem13 committed:
feat(data): Add mappers for Order and Contact_info

mohamedebrahem13 committed:
feat(data): Add StoreModel to Store entity mapper

mohamedebrahem13 committed:
feat(review): Add ReviewModel to Review entity mapper

mohamedebrahem13 committed:
Merge branch 'develop' into refactor/user

mohamedebrahem13 committed:
Merge remote-tracking branch 'origin/develop' into refactor/user

# Conflicts:
#	pubspec.lock

mohamedebrahem13 committed:
Merge remote-tracking branch 'origin/refactor/user' into refactor/user

# Conflicts:
#	pubspec.lock

mohamedebrahem13 committed:
fix pub file

mohamedebrahem13 committed:
remove id from mock

mohamedebrahem13 committed:
upgrade version

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

mohamedebrahem13 committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into refactor/user

# Conflicts:
#	lib/di/modules/repository_module.dart

mohamedebrahem13 committed:
refactor(address): Make address ID optional

mohamedebrahem13 committed:
refactor(home): Refactor `HomeAppBar` to fetch user data directly

mohamedebrahem13 committed:
refactor(user): Update `updateUserProfile` to accept more specific fields

mohamedebrahem13 committed:
refactor(user): Update User entity and model

This commit refactors the `User` entity, its corresponding `UserModel`, and the mapper to align with updated data requirements.

mohamedebrahem13 committed:
refactor(user): Update UserCubit to align with new user profile data structure

mohamedebrahem13 committed:
feat(profile): Enhance `UpdateUserProfileRequest` model

This commit expands the `UpdateUserProfileRequest` model to support additional user profile fields. The following optional `String` properties have been added:

- `phoneNumber`
- `city`
- `country`
- `avatarUrl`

mohamedebrahem13 committed:
refactor(di): Switch from mock to real repository implementations

This commit updates the dependency injection setup to use the real repository implementations instead of the mock ones.

mohamedebrahem13 committed:
refactor(user): Update User entity and mock implementation

mohamedebrahem13 committed:
fix(order): Handle nullable address ID in order creation

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

abdo-essam commented on lib/data/core/api/api_endpoints.dart:
```diff
@@ -19,11 +19,11 @@ class ApiEndpoints {
-  static String userAvatar(String userId) => '$apiVersion/user/$userId/avatar';
-  static String userAddress(String userId) => '$apiVersion/user/$userId/address';
-  static String userChangePassword(String userId) => '$apiVersion/user/$userId/change-password';
+  static String userById() => '$apiVersion/user/profile';
```
rename it to userProfile not userbyid

marah222 commented on lib/data/repositories/user_repository_impl.dart:
```diff
@@ -1,108 +1,95 @@
   @override
   Future<Result<User>> updateUserProfile({
-    String? fullName,
+    String? firstName,
```
I think we should start using params classes, instead of separated parameters,
like this
class AuthenticateParams {
  final String firstname;
  final String lastname;
  final bool password; .......
  
  what do you think?

abdo-essam commented on lib/data/mock/mock_data_generator.dart:
```diff
@@ -17,16 +17,25 @@ class MockDataGenerator {
 
   // User Mock Data
   static User generateUser({int index = 0}) {
```
I think we don't need mocks anymore.

mohamedebrahem13 replied:
for the user yes but there is method used in auth

abdo-essam commented on lib/data/mock/repositories_impl/mock_user_repository_impl.dart:
remove the mock repo for the user

mohamedebrahem13 committed:
refactor(user): Remove unused mock user repository

mohamedebrahem13 committed:
fix(user): Use userProfile endpoint for fetching user data

mohamedebrahem13 committed:
refactor(auth): Use userProfile endpoint to get current user

mohamedebrahem13 committed:
refactor(user): Rename `userById` endpoint to `userProfile`

abdo-essam commented on lib/presentation/screens/home/widgets/home_app_bar.dart:
```diff
@@ -7,59 +7,40 @@ import '../../../../core/design_system/constants/app_images.dart';
-  }
+    return BlocBuilder<UserCubit, UserState>(
+      builder: (context, state) {
+        String userName = 'Guest';
```
this initial value will not be here ('Guest')
will be in UserState in initial 
and will take it from localization not hard coded

abdo-essam commented on lib/di/injection_container.dart:
```diff
@@ -12,7 +12,7 @@ Future<void> init() async {
-  //RepositoryModule.register(sl);
-  MockRepositoryModule.register(sl);
+  RepositoryModule.register(sl);
+  // MockRepositoryModule.register(sl);
```
uncomment this so we still need the mock for the remind repos

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.

mohamedebrahem13 committed:
refactor(localization): Remove ARB files and localization logic

mohamedebrahem13 committed:
chore(di): Switch to mock repository module for dependency injection

mohamedebrahem13 committed:
feat(localization): Add Arabic and English localization files

ghost commented:
<!-- sellio-bot-autoupdate-comment -->
**[Sellio Bot] Auto-update Status**


✅ Branch successfully updated from `develop`.
